content/blog/ggplot2-styling/index.Rmd

```diff
@@ -0,0 +1,1141 @@
+
+```{r aspect_ratio}
+phi <- (1 + sqrt(5)) / 2
+p + theme(aspect.ratio = phi)
+```
+
+In the cases where a cohesive set of properties serves as a theme element, ggplot2 has `element_*()` functions.
+One of the simpler elements is `element_line()` and we can declare a new set of line properties as follows:
+
+```{r red_line}
+red_line <- element_line(colour = "red", linewidth = 2)
+red_line
+```
+
+These elements can then be given to the `theme()` function to assign these properties to a specific part of the theme, like the `axis.line` in this example.
+
+```{r red_axis}
+p + theme(axis.line = red_line)
+```
+
+Below is an overview of elements and some common places where they are used:
+
+```{r element_table, echo=FALSE}
+elems <- tibble::tribble(
+  ~ Element, ~ Description,
+  "`element_blank()`",   "Indicator to skip drawing an element.",
+  "`element_line()`",    "Used for axis lines, grid lines and tick marks.",
+  "`element_rect()`",    "Used for (panel) backgrounds, borders and strips.",
+  "`element_text()`",    "Used for (sub)titles, labels, captions.",
+  "`element_geom()`",    "Used to set default properties of layers.",
+  "`element_polygon()`", "Not used, but provided for reasons of extension.",
+  "`element_point()`",   "Not used, but provided for reasons of extension."
+)
+knitr::kable(elems)
+```
+
+In addition to these elements in ggplot2, extension packages can also define custom elements.
+Generally speaking, these elements are variants of the elements listed above and often have slightly different properties and are rendered differently.
+For example `ggtext::element_markdown()` is a subclass of `element_text()`, but interprets the provided text as markdown.
+It applies some html tags like `<b>` for bold and `<i>` for italic when rendering the text.
+Another example is `ggh4x::element_part_rect()` that can draw a subset of rectangle borders.
+
+```{r element_markdown}
+p +
+  labs(title = "<b>Fuel</b> <i>efficiency</i>") +
+  theme(
+    plot.title = ggtext::element_markdown(),
+    strip.background = ggh4x::element_part_rect(colour = "black", side = "b")
+  )
+```
+
+### Hierarchy and inheritance
+
+Most theme elements are hierarchical.
+At the root, they are broadly applicable and change large parts of the plot.
+At leaves, they are very specific and allow fine grained control.
+Travelling from roots to leaves, properties of theme elements are inherited from parent to child.
+Some inheritance is very direct, where leaves directly inherit from roots (for example `legend.text`).
+Other times, inheritance is more arduous, like for `axis.minor.ticks.y.left`: it inherits from `axis.ticks.y.left`, which inherits from `axis.ticks.y`, which inherits from `axis.ticks`, which finally inherits from `line`.
+Most often, elements only have a single parent, but there are exceptions so the inheritance of theme elements is not strictly a directed acyclic graph.
```

Comment:
This is not really true. A DAG can have multiple inheritance as long as it is not cyclic. So the theme is a DAG but not a tree